core/include/scipp/core/element/arithmetic.h

```diff
@@ -22,19 +22,26 @@ constexpr auto add_inplace_types =
              std::tuple<float, int64_t>, std::tuple<float, int32_t>,
              std::tuple<int64_t, bool>>;
 
-constexpr auto add_inplace_nan_types =
-    arg_list<double, float, std::tuple<double, float>,
-             std::tuple<float, double>>;
+template <class T> struct ValueType { using value_type = T; };
+template <class T> struct ValueType<ValueAndVariance<T>> {
+  using value_type = T;
+};
```

Comment:
What is wrong with using https://github.com/scipp/scipp/blob/b6fa0b96c739055a49d99cce34c6fd11783cacc5/core/include/scipp/core/value_and_variance.h#L22?

Reply:
see usage below - templated class and specialisation allow us to determine if we are dealing with underlying floating point types whether value or ValueAndVariance.

Reply:
I see. Have you considered doing what you did for other functions and check for ints:
```cpp
if (isInt(var.dtype())
  return sum(var);
return transform(var, nanplus);
```
or something like that?

Reply:
With Matt's suggestion below some `detail::isnan` supporting ints and floats we could simplify all the type handling here.

Reply:
Now unused?